### ISSUE TITLE
Crashes when an invalid (?) time filter is chosen

### ISSUE THREAD

**Issue by redstreet** [aumayr/beancount-web#47]
I tried to type "2014-2015" into the Time filter. It would be nice if that format worked (sneaking in a feature request here!), but given that it doesn't, I get an error:

http://localhost:5000/account/Assets:Bank:BankNane/monthly_balances/?filter_time=2014-2015

```python
builtins.TypeError
TypeError: 'NoneType' object is not iterable

Traceback (most recent call last)
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1836, in __call__
return self.wsgi_app(environ, start_response)
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1820, in wsgi_app
response = self.make_response(self.handle_exception(e))
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1403, in handle_exception
reraise(exc_type, exc_value, tb)
File "/usr/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
Open an interactive python shell in this frameraise value
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1817, in wsgi_app
response = self.full_dispatch_request()
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1477, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1381, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "/usr/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
raise value
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1473, in full_dispatch_request
rv = self.preprocess_request()
File "/usr/lib/python3.4/site-packages/flask/app.py", line 1666, in preprocess_request
rv = func()
File "/usr/lib/python3.4/site-packages/beancount_web-0.1.0a2-py3.4.egg/beancount_web/application.py", line 250, in perform_global_filters
payees=app.entry_filters.get('payees', set()).copy())
File "/usr/lib/python3.4/site-packages/beancount_web-0.1.0a2-py3.4.egg/beancount_web/api.py", line 238, in filter
self.apply_filters()
File "/usr/lib/python3.4/site-packages/beancount_web-0.1.0a2-py3.4.egg/beancount_web/api.py", line 207, in apply_filters
begin_date, end_date = parse_date(self.filters['time_str'])
TypeError: 'NoneType' object is not iterable
The debugger caught an exception in your WSGI application. You can now look at the traceback which led to the error.
To switch between the interactive traceback and the plaintext one, you can click on the "Traceback" headline. From the text traceback you can also create a paste of it. For code execution mouse-over the frame you want to debug and click on the console icon on the right side.

You can execute arbitrary Python code in the stack frames and there are some extra helpers available for introspection:

dump() shows all variables in the frame
dump(obj) dumps all that's known about the object
Brought to you by DON'T PANIC, your friendly Werkzeug powered traceback interpreter.
````

**Comment by yagebu** [comment]
This is basically a duplicate of #31 which I'll work on, but I'm quite busy right now, so it might take a few days before I get the time.

On the format: `2014 - 2015` will work. The spaces around the `-` are important, as this makes parsing the date simpler. Otherwise there can be confusion between `-` in a date or a range. Although it might be nice to remove this distinction.

**Comment by redstreet** [comment]
Good to know ranges already work!

**Comment by aumayr** [comment]
I close this issue as @yagebu will fix it in #31
